python/lsst/daf/butler/formatters/packages.py

```diff
@@ -0,0 +1,134 @@
+# This file is part of obs_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("PackagesFormatter",)
+
+import os.path
+
+from lsst.daf.butler.formatters.file import FileFormatter
+
+
+class PackagesFormatter(FileFormatter):
+    """Interface for reading and writing `~lsst.utils.packages.Packages`.
+
+    This formatter supports write parameters:
+
+    * ``format``: The file format to use to write the package data. Allowed
+      options are ``yaml``, ``json``, and ``pickle``.
+    """
+
+    supportedWriteParameters = frozenset({"format"})
+    supportedExtensions = frozenset({".yaml", ".pickle", ".pkl", ".json"})
+
+    # MyPy does't like the fact that the base declares this an instance
+    # attribute while this derived class uses a property.
+    @property
+    def extension(self) -> str:  # type: ignore
+        # Default to YAML but allow configuration via write parameter
+        format = self.writeParameters.get("format", "yaml")
+        ext = "." + format
+        if ext not in self.supportedExtensions:
+            raise RuntimeError(f"Requested file format '{format}' is not supported for Packages")
+        return ext
+
+    def _readFile(self, path, pytype):
```

Comment:
No type annotations?

Reply:
I will see if I can turn them on for this part of the hierarchy.